### ISSUE TITLE
How to debug more effectively?

### ISSUE THREAD

**Issue by ykhandler** [EliasKotlyar/Xiaomi-Dafang-Software#1]
Hi @EliasKotlyar , I've tried to compile v4l2rtspserver, but when i run it, i  got this error message:

----------------------------------------------------------------------------------------
root@Ingenic-uc1_1:mmcblk0p1]# ./v4l2rtspserver /dev/video0
1970-01-01 08:02:19,680 [NOTICE] - /test/v4l2rtspserver/src/main.cpp:548
Create V4L2 Source.../dev/video0
1970-01-01 08:02:19,681 [ERROR ] - src/V4l2Device.cpp:82
Cannot open device:/dev/video0 Operation not permitted
1970-01-01 08:02:19,682 [ERROR ] - src/V4l2Device.cpp:65
Cannot init device:/dev/video0
VIDIOC_STREAMOFF: Bad file descriptor
VIDIOC_REQBUFS: Bad file descriptor
----------------------------------------------------------------------------------------

Do you know how to debug it? So i can fix what's wrong. i've tried to copy the binary to /tmp folder and set chmod 777 and chown root,root the binary, but it remains. Also tried to chmod 777 to the /dev/video0.

**Comment by esarigul** [comment]
@EliasKotlyar Elias, I have mostly managed to get my camera up and running except audio, I used combination of openfang built root and your repository, when I use default build root audio feature (which supports oss audio) I dont hear anything even though dsp amd mixer are present in dev directory, then I cloned your software repo to compile driver which seems to have more options, I was able to compile both oss and oss2 kernel module but when I try to insert compiled module it says "can't insert: unkown symbol in module, or unknown parameter", I used kernel 3.10.14 I believe build root kernel source and your repos kernel directory are identical (correct me if I am wrong)
Bottom line how do I make audio work (in term of kernel menuconfig setting or separate compile for audio module)? thanks for your answer in advance
